### ISSUE TITLE
Plugin is ignoring `fetch` support for specific browser configuration

### ISSUE THREAD

**Issue by wesleycoder** [amilajack/eslint-plugin-compat#86]
The plugin keeps complaining about `fetch` support:
```
WARNING in ./src/helpers/api.js

/all/dev/integrations/src/helpers/api.js
  12:24  warning  fetch is not supported in Safari 10, Opera Mobile 12.1, Opera Mini all, iOS Safari 10.0-10.2, IE Mobile 10, IE 10, Blackberry Browser 7, Android Browser 4.4.3-4.4.4, Android UC Browser 11.4  compat/compat
```
My `package.json` has:
```json
{
  "browserslist": ["Chrome >= 47"]
}
```
And I have some code like this in `src/helpers/api.js`:
```js
export const api = {
	get: (url, params) => fetch('/endpoint').then(res => res.json())
};
```
As I'm targeting just Chrome 47 and above I should have no warnings about `fetch` which is supported since Chrome 42/43 according to https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API#Browser_compatibility.

**Comment by amilajack** [comment]
https://github.com/amilajack/eslint-plugin-compat/issues/72#issuecomment-288834016

**Comment by amilajack** [comment]
@wesleycoder are you still experiencing this issue?

**Comment by MaxMilton** [comment]
I'm also having this issue. No matter what I have for my `browserslist` compat will give a warning:
```
Promise.all() is not supported in Opera Mobile 12.1, Opera Mini all, IE Mobile 10, IE 10, Blackberry Browser 7 (compat/compat)
```
Example browserslist config (in `package.json`):
```
  "browserslist":  [
    "last 1 Chrome version"
  ]
```
My actual browserslist config:
```
  "browserslist": {
    "production": [
      "> 1% in AU",
      "> 1% in JP",
      "last 2 Chrome versions",
      "last 2 Firefox versions",
      "last 2 ChromeAndroid versions",
      "last 2 Edge versions",
      "last 2 iOS versions",
      "last 2 Safari versions",
      "not Explorer <= 11",
      "not ExplorerMobile <= 11"
    ],
    "development": [
      "last 1 Chrome version"
    ]
  }
```
both configs trigger the compat warning. For now I've just disabled compat :hear_no_evil:

**Comment by xiaofengqqcom123** [comment]
@MaxMilton Finally, how to solve this problem?

**Comment by MaxMilton** [comment]
@xiaofengqqcom123 I never did solve the problem. I've since stopped using this plugin at all.

**Comment by EnzoDomingues** [comment]
This is already documented at https://github.com/amilajack/eslint-plugin-compat/wiki/Adding-polyfills (linked to from the main page)

```
"settings": {
  "polyfills": ["promises"]
}
```
Adding ability for eslint-plugin-compat to detect that you've added a polyfill is very hard since that's something you would know only at run time. Eslint does static code analysis, not runtime. It would have to know/understand your build code which could involve gulp, grunt, webpack, browserify, etc. You could also have added it via for example polyfill.io.

**Comment by mikedeng** [comment]
In `v3` is should write as: 
```js
"settings": {
    "polyfills": ["Promise"]
}
```
